lang/ir/src/ir/item/event.rs

```diff
@@ -92,16 +92,17 @@ impl TryFrom<syn::ItemStruct> for Event {
                 "generic ink! event structs are not supported",
             ))
         }
-        match &item_struct.vis {
-            syn::Visibility::Inherited
-            | syn::Visibility::Restricted(_)
-            | syn::Visibility::Crate(_) => {
-                return Err(format_err_spanned!(
-                    &item_struct.vis,
-                    "non `pub` ink! event structs are not supported",
-                ))
-            }
-            _ => (),
+        let bad_visibility = match &item_struct.vis {
```

Comment:
Duplicate code fragment with `./storage.rs`. Not sure whether worth extracting, just flagging it.

Reply:
yeah not sure either. let me see if deduplication is easy.

Reply:
resolved. thanks for pointing out! duplicate code often happens during prototyping of new features etc. so this is super helpful!